proposals/NNNN-android-manager-listener-update.md

```diff
@@ -0,0 +1,253 @@
+        }
+    }
+    
+    // first we need to check if the SDL language matches or if we need to update configuration
+    DebugTool.logInfo("Proxy is connected. Initializing managers.");
+    initialize();
+}
+``` 
+
+### SdlManager.Updater
+
+The updater will be a nested class which behaves similar to the static `Builder` subclass.
+
+```java
+public class SdlManager {
+public class Updater {
+    private @Nullable String appName;
+    private @Nullable String shortAppName;
+    private @Nullable List<TTSChunk> ttsName;
+    private @Nullable List<String> voiceRecognitionCommandNames;
+
+    public Updater(String appName, String shortAppName, List<TTSChunk> ttsName, List<String> voiceRecognitionCommandNames) {
+        this.appName = appName;
+        this.shortAppName = shortAppName;
+        this.ttsName = ttsName;
+        this.voiceRecognitionCommandNames = voiceRecognitionCommandNames;
+        ChangeRegistration
+    }
+
+    public void setAppName(String appName) {
+        this.appName = appName;
+    }
+
+    public void setShortAppName(String shortAppName) {
+        this.shortAppName = shortAppName;
+    }
+
+    public void setTtsName(List<TTSChunk> ttsName) {
+        this.ttsName = ttsName;
+    }
+
+    public void setVoiceRecognitionCommandNames(List<String> voiceRecognitionCommandNames) {
+        this.voiceRecognitionCommandNames = voiceRecognitionCommandNames;
+    }
+
+    void update(Language language) {
+        ChangeRegistration request = new ChangeRegistration(language, language);
+        SdlManager.this.hmiLanguage = language;
+
+        if (appName != null) {
+            request.setAppName(appName);
+            SdlManager.this.appName = appName;
+        }
+
+        if (shortAppName != null) {
+            request.setNgnMediaScreenAppName(shortAppName);
+            SdlManager.this.shortAppName = shortAppName;
+        }
+
+        if (ttsName != null) {
+            request.setTtsName(ttsName);
+            SdlManager.this.ttsChunks = new Vector<>(ttsName);
+        }
+
+        if (voiceRecognitionCommandNames != null) {
+            request.setVrSynonyms(voiceRecognitionCommandNames);
+            SdlManager.this.vrSynonyms = new Vector<>(voiceRecognitionCommandNames);
+        }
+
+        SdlManager.this.sendRPC(request);
+    }
+}
+```
+
+## Potential downsides
+
+From a technical point of view there is no downside to the added features. However as Java interfaces force you to implement every single method which can be distracting. 
+
+## Impact on existing code
+
+The additional methods in the interface have impact to implementations of app developers and may be distracting. Besides of that there's no existing code changed.
+
+## Alternatives considered
+
+To avoid forcing app developers to implement the methods added to `SdlManagerListener` each method could be added to the SDL manager builder to allow the app developer to choose what methods are interesting to listen.
```

Comment:
Update to: 
To avoid forcing app developers to implement the methods added to `SdlManagerListener` each method could be added to the SDL manager builder to allow the app developer to choose what methods are interesting to listen **to**.